mmedit/models/backbones/sr_backbones/cain_net.py

```diff
@@ -0,0 +1,386 @@
+            nn.Conv2d(
+                mid_channels,
+                mid_channels // reduction,
+                1,
+                padding=0,
+                bias=True), nn.ReLU(inplace=True),
+            nn.Conv2d(
+                mid_channels // reduction,
+                mid_channels,
+                1,
+                padding=0,
+                bias=True), nn.Sigmoid())
+
+    def forward(self, x):
+        """Forward function for CALayer.
+
+        Args:
+            x (Tensor): Input tensor with shape (n, c, h, w).
+
+        Returns:
+            Tensor: Output tensor with shape (n, c, h, w).
+            Tensor: CA tensor with shape (n, c, 1, 1).
+        """
+
+        y = self.avg_pool(x)
+        y = self.channel_attention(y)
+        return x * y, y
+
+
+class RCA(nn.Module):
+    """Residual Channel Attention Module.
+
+    Args:
+        mid_channels (int): Channel number of the intermediate features.
+        kernel_size (int): Kernel size of RCA. Default: 3.
+        downscale (bool): Down scale or not. Default: False. Default: False.
+        reduction (int): Channel reduction of CA. Default: 16.
+        return_ca (bool): Return CA tensor or not. Default: False.
+        norm (None | function): Norm layer. If None, no norm layer.
+            Default: None.
+        act (function): activate function. Default: nn.ReLU(True).
+    """
+
+    def __init__(self,
+                 mid_channels,
+                 kernel_size=3,
+                 downscale=False,
+                 reduction=16,
+                 return_ca=False,
+                 norm=False,
+                 act=nn.ReLU(True)):
+        super().__init__()
+
+        self.body = nn.Sequential(
+            ConvNorm(
+                mid_channels,
+                mid_channels,
+                kernel_size,
+                stride=2 if downscale else 1,
+                norm=norm), act,
+            ConvNorm(
+                mid_channels, mid_channels, kernel_size, stride=1, norm=norm),
+            CALayer(mid_channels, reduction))
+        self.return_ca = return_ca
+        self.down_scale = downscale
+        if self.down_scale:
+            self.down_conv = nn.Conv2d(
+                mid_channels, mid_channels, kernel_size=3, stride=2, padding=1)
+
+    def forward(self, x):
+        """Forward function for RCA.
+
+        Args:
+            x (Tensor): Input tensor with shape (n, c, h, w).
+
+        Returns:
+            if return_ca:
+                Tensor: Output tensor with shape (n, c, h, w).
+                Tensor: CA tensor with shape (n, c, 1, 1).
+            else:
+                Tensor: Output tensor with shape (n, c, h, w).
+        """
+
+        res = x
+        out, ca = self.body(x)
+
+        if self.down_scale:
+            res = self.down_conv(res)
+        out += res
+
+        if self.return_ca:
+            return out, ca
+        else:
+            return out
+
+
+class ResidualGroup(nn.Module):
+    """Residual Group.
```

Comment:
Residual Group, consisting of a stack of residual channel attention, followed by a convolution.